### ISSUE TITLE
Upper deck breaches not repairable (Edit- Dual issues, visual obscured/ tile code with workaround)

### ISSUE THREAD

**Issue by SerynEngi** [BeeStation/NSV13#1189]
## Testmerges:

Not Relevant, probably.

## Reproduction:

Edit: Workaround is to right-click and specifically select the space tile. The visually obscured hole is another issue altogether.

This bug predates newnitions as far as I know, I remember attempting to fix it before hand, but kept forgetting the recording software until round 1733.

Step 1: Damage the ship in such a manner that some of the upper tiles external to the internal compartments are partially damaged.
Step 2: Attempt to fix holes outside of compartments with RCD
Step 3: Fall in hole when RCD does nothing.
Step 4: After coming back up to the other deck, attempt to lay down rods.
Step 5: When that doesnt work, and you fall down, go back up to step 6 (not demonstrated in the video).
Step 6: Throw a metal foam grenade next to known location of the hole. Watch nothing happen to the hole. Fall in the hole once more.

In short, it becomes impossible to repair plating on the upper deck that isnt in a compartment, leading to holes that are not visually there and can't be fixed by any method, be it the use of an RCD or manually attempting to use rods and then tile it as demonstrated in the video.

Not demonstrated is the use of metal foam grenades also failing to work. This appears to apply specifically to plating outside normally habitable locations such as the flight deck.

If for whatever reason, the git or the browser decides to not let you see the video on your browser, poke me on discord, the video is sufficiently small to pass it through.

https://user-images.githubusercontent.com/75588150/105647285-62588200-5e59-11eb-93d0-ac652c77166e.mp4

**Comment by Bokkiewokkie** [comment]
Alright so this isn't really a bug, it's just a matter of knowing how things work.
Due to the way hull plating works, it's not a turf but a structure. This means that it's on the layer (order of displaying/interacting) above turfs (floors/walls) and will be hit before you hit the actual floor by just clicking on it.
To prevent things like this from _actually_ breaking interaction with certain things, it's possible to just alt-click on the tile to show all the contents that are there! After that you can just simply apply your fixings to the open space/ lattice in the [Floor] tab that has now opened in your game window on the top right, and go on with your day.

P.S.: Metal foam grenades and RCD's don't like working together with lattices over open spaces, which is probably why the grenades didn't work.

**Comment by SerynEngi** [comment]
This has been partially solved, as metal foam does appear to actually work now. A fix for the plating visuals may be to code it so that it does not display the plating tile if the floor tile beneath it has been destroyed.

**Comment by SerynEngi** [comment]
While it is still possible to fall through due to visual layering, this does appear to have been largely fixed.
